### PR TITLE
[WIP] Fixes for the tendermint implementation

### PR DESCRIPTION
**Pull request by JekaMas**
fixes https://github.com/clearmatics/autonity/pull/116

### CODE REVIEW

File: helloworld/docker-compose.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ services:
     volumes:
       - ./nodekey1:/nodekey:ro
       - ./keystore:/keystore:ro
-      - ./genesis-ibft.json:/genesis.json:ro
+      - ./genesis-tendermint.json:/genesis.json:ro
```
Comment (aneequesafdar) [inline helloworld/docker-compose.yml]: Can we please not change the helloworld dir, unless there is a specific reason to.

File: consensus/tendermint/types.go [hunk 2]
```diff
@@ -70,7 +70,11 @@ func (sub *Vote) EncodeRLP(w io.Writer) error {
 
 // DecodeRLP implements rlp.Decoder, and load the consensus fields from a RLP stream.
 func (sub *Vote) DecodeRLP(s *rlp.Stream) error {
-	var subject = new(Vote)
+	var subject struct {
+		Round             *big.Int
+		Height            *big.Int
+		ProposedBlockHash common.Hash
+	}
```
Comment (aneequesafdar) [inline consensus/tendermint/types.go]: How does the above change fix an infinite loop? This error will also be present in other DecodeRLP methods

Comment (aneequesafdar) [reply consensus/tendermint/types.go]: I can update the rest

Comment (JekaMas) [reply consensus/tendermint/types.go]: Nope. It introduces a new type without DecodeRLP.
Without it the DecodeRLP will be called until stack will be overflowed.

File: helloworld/docker-compose.yml (hunk 1 above)
Comment (JekaMas) [reply helloworld/docker-compose.yml]: Ok, it was just for debug.

**Pull request merged**